framework/src/problems/FEProblemBase.C

```diff
@@ -6091,6 +6091,9 @@ FEProblemBase::meshChangedHelper(bool intermediate_change)
   // repartitioning done in EquationSystems::reinit().
   _mesh.meshChanged();
 
+  for (const auto & mci : _notify_when_mesh_changes)
+    mci->meshChanged();
```

Comment:
It would be great if you could put some code comment here on why this has to be where it is